### ISSUE TITLE
libnd4j: batchnorm op needs axis arg (NCHW format support)

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/deeplearning4j#6423]
Required for import, and later use in DL4J
https://www.tensorflow.org/api_docs/python/tf/layers/BatchNormalization
https://github.com/deeplearning4j/deeplearning4j/blob/master/libnd4j/include/ops/declarable/generic/nn/batchnorm.cpp

TF implementation supports an axis arg, but our libnd4j batch norm implementation does not.
For example, for CNNs, we have:
NHWC: axis = 3 (works with the current 'must be broadcastable' check)
NCHW: axis = 1 (not possible in our implementation)

There's also the question of supporting multiple axes. TF batch norm supports this, but I can't think of a case where this is actually useful in practice. i.e., batch norm is always applied along 1 dimension for all of MLPs (2d input), RNNs/CNN1D (3d input), CNNs (4d input) and 3D CNNs (5d input) - along features or channels in each case.
In summary, I think we should add support for 1 axis arg only for now, unless adding support for multiple axis args is not much more work.

**Comment by shyrma** [comment]
Hi Alex

I have provideв multi-axes arg in batchnorm op.
However I implemented this as new op (name is batchnorm_new).
I didn't delete/rewrite our previous batchnorm op, since it have another input validation and corresponds to following tf op:
https://www.tensorflow.org/api_docs/python/tf/nn/batch_normalization
whereas 
batchnorm_new corresponds to 
https://www.tensorflow.org/api_docs/python/tf/layers/BatchNormalization

the main difference between them is: for batchnorm rank of input array  can be < rank of mean/variance/gamma/beta, and mean/variance/gamma/beta all can have different rank;
for batchnorm_new input rank >= rank of mean/variance/gamma/beta always
and mean/variance/gamma/beta all should have identical shape

**Comment by AlexDBlack** [comment]
Finally got to using the new op - looks like this works well - thanks!
